bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/texteditor/codemining/ZeroWidthSpaceLineContentCodeMiningProvider.java

```diff
@@ -0,0 +1,39 @@
+package org.eclipse.ui.internal.texteditor.codemining;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+
+import org.eclipse.jface.text.ITextViewer;
+import org.eclipse.jface.text.Position;
+import org.eclipse.jface.text.codemining.AbstractCodeMiningProvider;
+import org.eclipse.jface.text.codemining.ICodeMining;
+
+public class ZeroWidthSpaceLineContentCodeMiningProvider extends AbstractCodeMiningProvider {
+
+	private static final char ZWSP_SIGN = '\u200b';
```

Comment:
we should also add
```
\u200c  zero width non-joiner
\u200d  zero width joiner
\ufeff  zero width no break space
```
See: https://plugins.jetbrains.com/plugin/7448-zero-width-characters-locator